cmake/modules/AddSwift.cmake

```diff
@@ -282,6 +282,10 @@ function(_add_host_variant_c_compile_flags target)
       ${${SWIFT_HOST_VARIANT_ARCH}_INCLUDE})
     target_compile_definitions(${target} PRIVATE
       __ANDROID_API__=${SWIFT_ANDROID_API_LEVEL})
+    if(SWIFT_HOST_VARIANT_ARCH STREQUAL armv7 AND SWIFT_ANDROID_API_LEVEL GREATER_EQUAL 24)
+      target_compile_definitions(${target} PRIVATE
+        _FILE_OFFSET_BITS=64)
```

Comment:
This took me a while to track down. The issue is that [LLVM automatically adds this define for 32-bit Android when the API level is 24 or higher](https://github.com/llvm/llvm-project/blob/b36bdfe5ca0c2b863248f327b03d41516b38dc11/llvm/cmake/modules/HandleLLVMOptions.cmake#L342), as it is when I'm building. However, since this define wasn't applied here in the Swift frontend, it caused issues where the frontend would expect a different layout of the `clang::Module` object when used in `ClangImporter` than the LLVM backend, causing problems where Macro literals from C headers wouldn't be visible in Swift.
Took me some time running the Swift compiler in lldb to figure out that the backend was using a different offset into the object for `NameVisibility` than the frontend, which adding this define corrects.

Reply:
Ewww that is terrible.  Would you mind adding a comment with a link to the LLVM CMake?  This is subtle and sounds horrible to track down.

Reply:
Good idea.

Reply:
Btw, LLVM applies this define for all 32-bit platforms, not just 32-bit Android. Should I break this out of the Android block and do the same here? I was just narrowly fixing my Android ARMv7 problem, but this will likely break linux ARMv7 and other 32-bit platforms too.

Reply:
Yes, please, could just base that on CMAKE_SIZEOF_VOID_P EQUALS 4`?  That sounds like the right thing to do to avoid breaking this on all the other platforms.

Reply:
Looking at the full diff, this seems not possible yet :-(.